### ISSUE TITLE
508-defect-2 [COGNITION]: Find Form Results - Spanish content MUST include a lang="ES" attribute for proper pronunciation with assistive technology

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#18434]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh, Trevor_

## User Story or Problem Statement

As a screen reader user who prefers to hear form instructions in Spanish, I want my assistive technology to read instructions with the right pronunciation and cadence.

## Details

This issue has been documented before at the [part level in 18031](https://github.com/department-of-veterans-affairs/va.gov-team/issues/18031) and the [page level in 14432](https://github.com/department-of-veterans-affairs/va.gov-team/issues/14432). Documenting this again for awareness that it must be adjusted in find form results too.

Spanish content is not read accurately when the language attribute is defined for English, as we do on all our pages' HTML element. We should add `lang="es"` attributes to the P tags that wrap Spanish language instructions on our find form detail results.

## Acceptance Criteria

- [ ] All Spanish content is identified with a `lang="es"` attribute
- [ ] Zero axe violations with this updated markup
- [ ] Screen readers change pronunciation and shift into Spanish when they encounter properly tagged text. This is usually cued to users by a short pause, and a different tone and inflection of the screen reader voice.

## Environment

* https://staging.va.gov/find-forms/?q=10-10ez 

### Before
```html
<a href="https://www.va.gov/find-forms/about-form-10-10ez-espanol?_ga=2.256309000.249431449.1610379387-1042145308.1601927343" class="vads-u-text-decoration--none">
  <dfn>
    <span class="vads-u-visibility--screen-reader">Visit the landing page for Form number</span>
    10-10EZ (esp) 
  </dfn>
  Instrucciones y Solicitud de Beneficios Medicos
  <i aria-hidden="true" role="presentation" class="fas fa-angle-right vads-u-margin-left--0p25" style="vertical-align: middle;"></i>
</a>
```

### After
```diff
<a href="https://www.va.gov/find-forms/about-form-10-10ez-espanol?_ga=2.256309000.249431449.1610379387-1042145308.1601927343" class="vads-u-text-decoration--none">
  <dfn>
    <span class="vads-u-visibility--screen-reader">Visit the landing page for Form number</span>
    10-10EZ (esp) 
  </dfn>
+  <span lang="es">Instrucciones y Solicitud de Beneficios Medicos</span>
  <i aria-hidden="true" role="presentation" class="fas fa-angle-right vads-u-margin-left--0p25" style="vertical-align: middle;"></i>
</a>
```

## WCAG or Vendor Guidance (optional)

* [Language of Parts: Understanding SC 3.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-other-lang-id.html)
* [VSP documentation on accessibility for localization](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/resources/localization/accessibility-for-localization.md)

## Screenshots or Trace Logs

<img width="1462" alt="Screen Shot 2021-01-12 at 4 39 03 PM" src="https://user-images.githubusercontent.com/14154792/104377698-e1ae9300-54f4-11eb-88e1-f5ad6d764baf.png">

**Comment by zacharymorel** [comment]
They're may or may not be an issue with Language preference within the React apps. 
I will chat with some of my associates to see if they have a possible solution!

**Comment by joshkimux** [comment]
@zacharymorel Awesome! Thanks so much!

**Comment by zacharymorel** [comment]
@joshkimux What does the actual translation of from English to Spanish? 

Does the screen reader do that or is it does the user use google to translate the page?

**Comment by joshkimux** [comment]
@zacharymorel Great question! It's not used to _translate_ per say so much as it is to ensure the screen reader is pronouncing the words correctly. More on that in [SC 3.1.2: Language of Parts](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html): "The human language of each passage or phrase in the content can be programmatically determined except for proper names, technical terms, words of indeterminate language, and words or phrases that have become part of the vernacular of the immediately surrounding text."

Or, basically, it signals screen readers pronunciation engines to switch to another language

**Comment by MarciMcGuireGCIO** [comment]
Marking this as blocked since there's a larger sitewide effort by the Globalization team to address language.

**Comment by MarciMcGuireGCIO** [comment]
Moving all outstanding Find a VA Form 508 issues into a single Epic to clean up backlog.

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel, I'm taking this one out of `blocked` since it seems very similar to the change you just made on the Detail pages.  LMK if it's not.  Thx!

**Comment by zacharymorel** [comment]
I can do a look soon to see if we have the correct data for it!

**Comment by zacharymorel** [comment]
@MarciMcGuireGCIO after looking at the lighthouse Forms API data response, the api is not ready. It's **language** attribute needs to reflect what the Drupal cms exposes in their _fieldVaFormLanguage_ field.

**Comment by zacharymorel** [comment]
@joshkimux This is now live on Staging & Prod.
Can you review this when you get a chance?

**Comment by joshkimux** [comment]
@zacharymorel  just wrapped up testing this and came across some odd screen reader behavior on VO. You can [test for yourself using this codepen I made too](https://codepen.io/joshkimux/pen/vYgVWxN) 😄 

## Notes and recommendations

A `span` with the `lang` attribute within an `a` will have its `lang` attribute ignored. It will only be pronounced with the appropriate accent when:
- The `<span lang="es">` is _not nested_ within the `a` tag OR
- The `a` tag holds the `lang` attribute itself e.g. `<a lang="es">`

To fix this, I recommend:
-  adding the`lang="es"` attribute to the `a`
- removing the English `sr-only` text to simplify the code, reduce confusion, and reduce verbosity
- removing all the `span`s as they will no longer be needed if the `a` holds the `lang` attribute

### Current
```html
<a href="https://www.va.gov/find-forms/about-form-10-10ez-espanol"> 
  <span class="sr-only">Visit the landing page for Form number</span>
  10-10EZ (esp) 
  <span lang="es">Instrucciones y Solicitud de Beneficios Medicos</span>
</a>
```

### Recommendation

```html
  <a lang="es" href="https://www.va.gov/find-forms/about-form-10-10ez-espanol"> 
     10-10EZ (esp) Instrucciones y Solicitud de Beneficios Medicos
  </a>
```

**Comment by zacharymorel** [comment]
@joshkimux , I am for removing all spans with the text `Visit the landing page for Form number`.  Is that cool with you?

**Comment by joshkimux** [comment]
@zacharymorel sorry for the late response! Yes, we can also delete the span wrapping the form name itself and move the `lang` attribute up to the `a`.

**Comment by zacharymorel** [comment]
No worries, I can make the adjustment!

**Comment by zacharymorel** [comment]
@joshkimux is it safe to say that having the Icon, which is hidden from the SR, is okay within the anchor tag? 
OR do I have to move it out of the anchor tag?

![Screen Shot 2021-04-21 at 10 11 51 AM](https://user-images.githubusercontent.com/26075258/115568129-0059be00-
a28a-11eb-90d8-74dca0a96d3c.png)

**Comment by joshkimux** [comment]
@zacharymorel Completely fine having it inside the anchor tag, it shouldn't be read out since it has `aria-hidden="true"` on it 😄

**Comment by zacharymorel** [comment]
PR open

**Comment by zacharymorel** [comment]
@joshkimux looks like this is on Prod now if you want to give it a whirl!

**Comment by joshkimux** [comment]
Confirmed this particular case has been fixed! Will write a new ticket for a new issue we identified
